bjoern/_bjoernmodule.c

```diff
@@ -3,16 +3,42 @@
 #include "wsgi.h"
 #include "filewrapper.h"
 
+#ifdef WANT_STATSD
+#include "statsd-client.h"
+#endif
+
 static PyObject*
 run(PyObject* self, PyObject* args)
 {
   ServerInfo info;
 
   PyObject* socket;
 
-  if(!PyArg_ParseTuple(args, "OO:server_run", &socket, &info.wsgi_app)) {
+#ifdef WANT_STATSD
+  info.statsd = NULL;
+  int statsd_enabled;
+  char* statsd_host;
+  int statsd_port;
+  char* statsd_ns;
+  char* statsd_tags = NULL;
+
+  if(!PyArg_ParseTuple(args, "OOiziz|z:server_run", &socket, &info.wsgi_app,
+                       &statsd_enabled, &statsd_host, &statsd_port, &statsd_ns, &statsd_tags)) {
+    return NULL;
+  }
+#else
+  char* ignored_str = NULL;
+  int ignored_int = 0;
+
+  if(!PyArg_ParseTuple(args, "OO|izizz:server_run", &socket, &info.wsgi_app, &ignored_int,
+                       &ignored_str, &ignored_int, &ignored_str, &ignored_str)) {
     return NULL;
   }
+  if (ignored_str != NULL || ignored_int != 0) {
+    PyErr_Format(PyExc_TypeError, "Unexpected statsd_* arguments (forgot to compile with statsd support?)");
```

Comment:
This can be printed as a warning and bjoern can still boot. Ignoring these arguments when statsd support is not compiled will make it easier to deploy different versions of bjoern without having to make changes in code.

Reply:
Explicit is better than implicit.

Ignoring arguments (even with warning) is always a terrible idea. Better be explicit about it and throw an error. If you have multiple different versions of bjoern running then this is complexity in your code, not in bjoern, so it should also be dealt with in your code.